contracts/ERC20AssetHolder.sol

```diff
@@ -0,0 +1,104 @@
+pragma solidity ^0.5.11;
+pragma experimental ABIEncoderV2;
+import './Outcome.sol';
+import './AssetHolder.sol';
+
+contract IERC20 {
+    // Abstraction of the parts of the ERC20 Interface that we need
+    function transfer(address to, uint256 tokens) public returns (bool success);
+    function transferFrom(address from, address to, uint256 tokens) public returns (bool success);
+}
+
+contract ERC20AssetHolder is AssetHolder {
+    address AdjudicatorAddress;
+    address TokenAddress;
+
+    constructor(address _AdjudicatorAddress, address _TokenAddress) public {
+        AdjudicatorAddress = _AdjudicatorAddress;
+        TokenAddress = _TokenAddress;
+    }
+
+    modifier AdjudicatorOnly {
+        require(msg.sender == AdjudicatorAddress, 'Only the NitroAdjudicator is authorized');
+        _;
+    }
+
+    IERC20 _token = IERC20(TokenAddress);
+
+    // **************
+    // Asset Management
+    // **************
+
+    function deposit(bytes32 destination, uint256 expectedHeld, uint256 amount) public {
+        require(_token.transferFrom(msg.sender, address(this), amount), 'Could not deposit ERC20s');
+
+        uint256 amountDeposited;
+        // This protects against a directly funded channel being defunded due to chain re-orgs,
```

Comment:
Maybe reword: this allows participants to reduce the wait between deposits, while protecting them from losing funds by depositing too early. Specifically it protects against the scenario:
1. Participant A deposits
2. Participant B sees A's deposit, which means it is now safe for them to deposit
3. Participant B submits their deposit
4. The chain re-orgs, leaving B's deposit in the chain but not A's

Reply:
Have overwritten with your explanation (e7a7aaf)